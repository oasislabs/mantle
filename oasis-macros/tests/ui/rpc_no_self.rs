oasis_std::contract! {

#[derive(Contract, Default)]
pub struct Counter(u32);

impl Counter {
    pub fn new(ctx: &Context) -> Result<Self> {
        Ok(Default::default())
    }

    pub fn incr(self, ctx: &Context) -> Result<()> {
        self.0 += 1;
        Ok(())
    }
}

}

fn main() {}
