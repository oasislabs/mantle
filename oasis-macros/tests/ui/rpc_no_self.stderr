error: First argument to `Counter::incr` should be `&self` or `&mut self`.
  --> $DIR/rpc_no_self.rs:11:17
   |
11 |     pub fn incr(self, ctx: &Context) -> Result<()> {
   |                 ^^^^

error: aborting due to previous error

